### ISSUE TITLE
Proposal: Implement extra_lazy for collection->getKeys() 

### ISSUE THREAD

**Issue by shandyDev** [doctrine/doctrine2#5758]
Hi, Guys!
I want propose make for all extra_lazy associations, something like this:
```
$sql = 'SELECT indexBy FROM ' . $quotedJoinTable . ' WHERE ' . implode(' AND ', $whereClauses);
return (array) $this->conn->fetchColumn($sql, $params, 0, $types);
```
This should return an array IDs by simple SQL instead loop (and hydrate!) all collection.

**Comment by DHager** [comment]
Makes sense, although I've avoided keyed-collections because AFAIK when you update items (or flush them to get an auto-generated ID) the association-keys may be stale or inconsistent.
